### ISSUE TITLE
A link contains no text

### ISSUE THREAD

**Issue by shiffman** [CodingTrain/website#791]
Also on homepage:

<img width="622" alt="screen shot 2018-10-19 at 2 36 05 pm" src="https://user-images.githubusercontent.com/191758/47237214-61983200-d3ac-11e8-85ed-5a67935293df.png">

<img width="295" alt="screen shot 2018-10-19 at 2 35 49 pm" src="https://user-images.githubusercontent.com/191758/47237222-64932280-d3ac-11e8-801f-54674e7874f7.png">

**Comment by nilsweb** [comment]
Same as #790 but with the header image. Any suggestions?

**Comment by shiffman** [comment]
Yes, I am not sure of best practice here. Some ideas:

* alt text?
* aria-hidden attribute? [aria attributes](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) are special accessibility attributes for screen readers.
* text with display set to none?

I think this comment applies to #790 too?

**Comment by nilsweb** [comment]
> alt text?

That wouldn't really work here since it is just a background image (which is also not perfect but works without having to deal with a whole bunch of other issues).

> aria-hidden attribute? [aria attributes](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) are special accessibility attributes for screen readers.

That might be a good idea. I'll look into that!

> text with display set to none?

I'm not quite sure if screen readers will just skip that then since it shouldn't be visible to the user (not _seen_ and therefore also not _read_)?

> I think this comment applies to #790 too?

Yes!

**Comment by shiffman** [comment]
@mithru do you have any thoughts or advice?

**Comment by mithru** [comment]
I ran this locally and used the `title` attribute. Voice Over reads out "Home page" in the example below:
`<a href="/" title="Home page"> <div class="header-image"></div> </a>`

I'm not sure if this will resolve the issue on WebAIM's checker (I can't check as it doesn't work locally). I'll try this one one of my public sites and get back. If you try this before I do, can you let me know if the checker still complains? 

Responses to the three suggestions above:
1. Alt text in this case might not be the best considering it's a link to the home page (which, I believe, is important in terms of navigation). Also, the image here seems more decorative than informational? I understand the visually welcoming tone of the coding train banner, and that we might want to translate that tone to everyone, but I'm personally more convinced that this image is more decorative than informational. So making screenreaders avoid it might be the way forward. 

2. aria is great in general, but I wouldn't have screenreaders avoid this link if other users are able to navigate to the home page by clicking it. 

3. Text with display set to none will actually work! Screenreaders read out elements with `display: none`. As a general note, as far as I understand, screenreaders are not affected by css. If my suggestion above doesn't resolve the issue on wave / other accessibility checkers, using this certainly should. Let me know either way. I'll report back if I manage to check it first.

**Comment by mithru** [comment]
not sure how github notifies people about edits to comments, but just wanted to point out that I edited point 3 in my previous comment.

**Comment by nilsweb** [comment]
Thanks for your advice @mithru!

I've read that the `title` attribute might sometimes be skipped by screen readers but I couldn't test it either. What checker did you use @shiffman?

The [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute) attribute might do the trick? This then could be used for the home page link and also the hamburger on mobile.

Accessibility seems to be one of those topics where you encounter a lot of "might", "possibly" and "sometimes" and nobody really knows how it works out in the end...

**Comment by phucduong86** [comment]
Is there any reasons why you are not using:
```
<a href="/">
  <img src="_<link to image>_" alt="CodingTrain Homepage">
</a>
```

It would probably be the most semantic if you want to keep the background image as the link.
That being said the ideal accessibility improvement would be leaving the background image as presentation and adding a "Home" link to the Navigation bar, that way all the navs are grouped together.

As a link image right now, when keyboard user tab focus into it, it is also very difficult to see where the focus is, means it still fail the Perceivable guideline.

**Comment by nilsweb** [comment]
I haven't used an `img` tag because it's not that great to handle scaling for multiple screen sizes. With a background image on the `div` you just have to set one property and it works and since it's not content-related (it just "looks nice") it shouldn't be a problem.

That's actually a really simple but effective idea... I haven't even thought of that... 😂 This would in theory solve the issue and make it even more accessible since it would also be part of the `nav`.

**Comment by bjorvack** [comment]
I mostly use image replacement. (Move the text away from the screen)
A screen reader is capable off reading this

**Comment by phucduong86** [comment]
Hi,

Can you please elaborate on this use case? I am interested to know :)

**Comment by bjorvack** [comment]
It's mostly used for logo's and banners with text or images in a news article. Instead off using an image tag you use a background image. Inside the tag you could place a description of the image, where it go's to ect.

**Comment by phucduong86** [comment]
If you want to use the img as background image it is a great solution. It does get you around the screen reader issue but not so great for the visually enabled keyboard users. Logos are better candidate for this use case because at least browser would give it the default focus ring styling. Banners have this focus styling ring as well but the top and 2 side borders are hidden in the view port, which left a faint border at the bottom of the banner and is very difficult for user to distinguish. 
Experience web users are more likely to assume logo and background image are linked to the homepage. This is largely due to the fact that websites are designed that way historically. However it is not intuitively intepreted by a new web user and a consistently styled navigation bar would greatly increase the site's usability in general.
Sorry about the long read please let me know if anyone wants to discuss this topic further outside of this context. :)

**Comment by Fraasi** [comment]
I think specifically using [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute) attribute should solve this issue?

**Comment by Fraasi** [comment]
This was propably solved in #1318 
@phucduong86 can you  review if this is still an issue or could be closed?

**Comment by phucduong86** [comment]
Yeeah this cab be closed :)
